src/js/base/pyret-tokenizer.js

```diff
@@ -119,9 +119,19 @@ define("pyret-base/js/pyret-tokenizer", ["jglr/jglr"], function(E) {
 
   const number = new RegExp("^[-+]?" + unsigned_decimal_part, STICKY_REGEXP);
 
-  const badNumber = new RegExp("^~?[+-]?\\.[0-9]+(?:[eE][-+]?[0-9]+)?", STICKY_REGEXP);
+  const unsignedRationalorDecimal = "(?:" + unsigned_rational_part + "|" + unsigned_decimal_part + ")";
 
-  const roughnum = new RegExp("^~[-+]?"  + "(?:" + unsigned_rational_part + "|" + unsigned_decimal_part + ")", STICKY_REGEXP);
+  const numberwithoutIntegerPart = "^~?[+-]?\\.[0-9]+(?:[eE][-+]?[0-9]+)?";
+
+  const nonRoughnumwithDots = "^[+-]?" + unsignedRationalorDecimal + "\\.\\.+";
```

Comment:
Yes. I tried making it one or more, with a negative lookahead to make sure it isn't followed by a digit, but there is a test `5.{1}` that is supposed to elicit a different error.